### ISSUE TITLE
Workflow for contributors (changelog)

### ISSUE THREAD

**Issue by thiagotalma** [cantino/huginn#1549]
Hello @cantino,

I actively participate in various open source projects and, although not to program in Ruby, I follow the Huginn closely whenever I can try to help improve it.

But I miss a better control of the changes that occur in the project. Lack a well-defined process to record what was done.

One example that I think it would fit very well here in Huginn is Yii2 project, a PHP framework.

See how the collaborative process:
https://github.com/yiisoft/yii2/blob/master/docs/internals/git-workflow.md

I especially like the part of the changelog:
https://github.com/yiisoft/yii2/blob/master/docs/internals/git-workflow.md#5-update-the-changelog

What do you think of better organize the work around here? A bit of bureaucracy does not hurt  ;)

**Comment by cantino** [comment]
Hey @thiagotalma,

I definitely want to make sure we have good documentation and process to make development easy for everyone.  Are you suggesting a new document, or simply that we encourage people to add themselves to the [changelog](https://github.com/cantino/huginn/blob/master/CHANGES.md)?

**Comment by thiagotalma** [comment]
I do not see much need for a document describing the process.

But I really miss the organized changelog.
When we pull the project, we know exactly what changes are receiving.

The format adopted by Yii2 is very good. The changelog says what was done, who did and in which PR was done.

**Comment by cantino** [comment]
I'd be supportive if you want to change the CHANGELOG to be an ordered list of PR numbers without dates, so that updating it is easier.  Do you know if merge conflicts are common with everyone putting lines at the top?

**Comment by dsander** [comment]
@cantino Yeah that will generate conflicts for every PR that changes the CHANGELOG and is not on top of master, noticed that issue while contributing to GitLab.
